### PR TITLE
Spinnaker Chart

### PR DESCRIPTION
**Pull request by viglesiasce**
This installs [Spinnaker](http://spinnaker.io) and its dependencies as well as configures Spinnaker to operate as a deployment engine for the local Kubernetes cluster. It also configures 2 Jenkins jobs for use in the implementation of CI pipelines:

1. build-image: Build and push a Docker image
2. run-script: Clone a repository and then run a command

For more info about Spinnaker on Kubernetes, check out:
http://www.spinnaker.io/docs/kubernetes-source-to-prod

### CODE REVIEW

**Comment by viglesiasce** [comment]
@k8s-bot e2e test this

**Comment by viglesiasce** [comment]
@technosophos @michelleN here is the chart that I was saying takes a really long time to install and sometimes causes timeouts. 

The issue is that it requires multiple job hooks that depend on certain pods to be fully running. The hook jobs do retry and back off but helms' default timeout is not enough for a freshly installed cluster to pull down all of the images and such in time.

Is there a way to make sure Helm does not time out while post install hooks are running or furthermore to ensure that post install hooks have their own timeout?

**Comment by lachie83** [comment]
@viglesiasce great contrib. I've worked with spinnaker in the past and would like to give the review a shot. It's quite the chart...

**Comment by technosophos** [comment]
In 2.2 there is a `helm install --timeout=INT` flag that can be set on the CLI.

File: stable/spinnaker/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+Redis and Minio are used as the stores for Spinnaker state.
+
+For more information on Spinnaker and its capabilities, see it's [documentation](http://www.spinnaker.io/docs).
+
+## Installing the Chart
+
+To install the chart with the release name `my-release`:
+
+```bash
+$ helm install --name my-release stable/spinnaker
```
Comment (lachie83) [inline stable/spinnaker/README.md:18]: Mention the timeout value on install

File: stable/spinnaker/templates/NOTES.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+1. You will need to create 2 port forwarding tunnels in order to access the Spinnaker UI:
+  export GATE_POD=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "component=gate,app={{ template "fullname" . }}" -o jsonpath="{.items[0].metadata.name}")
+  export DECK_POD=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "component=deck,app={{ template "fullname" . }}" -o jsonpath="{.items[0].metadata.name}")
+  kubectl port-forward $GATE_POD 8084 &
```
Comment (lachie83) [inline stable/spinnaker/templates/NOTES.txt]: Please add `--namespace {{ .Release.Namespace }}` to both port forwards

File: stable/spinnaker/README.md (hunk 1 above)
Comment (viglesiasce) [reply stable/spinnaker/README.md:18]: It won't exist in helm until 2.2

File: stable/spinnaker/templates/NOTES.txt (hunk 2 above)
Comment (viglesiasce) [reply stable/spinnaker/templates/NOTES.txt]: Done deal.

File: stable/spinnaker/README.md (hunk 1 above)
Comment (viglesiasce) [reply stable/spinnaker/README.md:18]: Added a note for now. Will update when helm 2.2 ships

**Review by lachie83 — APPROVED** [review]
LGTM

**Pull request merged**